explainer.md

```diff
@@ -1,624 +1,397 @@
+
+### Mixing in parts of Page-B and associating them with Page-A parts
+
+At this stage, Page-B identifies parts of its own page to be involved in the transition. This happens in the same way as part 1 with one difference: The textures and styles from Page-B will be updated if the underlying page updates. This means things like animated gifs will play, rather than being frozen on whatever frame they were on when they were captured.
+
+The developer can associate particular parts from Page-A to parts from Page-B. This would usually be done if they're equivalent. In this case, the headers, share buttons, and roots are equivalent. When this happens, the texture from the Page-B capture is added to the transition part:
 
-```html
-<html>
-<head>
-  <style>
-    body {
-      background-color: blue;
-    }
-
-    .animated {
-      position: relative;
-      top: 300px;
-      width: 100px;
-      height: 100px;
-      background-color: red;
-
-      border: 10px solid black;
-      border-radius: 10% 10%;
-      box-shadow: 0px 0px 10px;
-    }
-
-    .animated:transition {
-      /* Retain the border to ensure it is painted transparent but the box size is unchanged. */
-      border: 10px solid transparent;
-      border-radius: none;
-      box-shadow: none;
-    }
-
-    ::shared-container(header-id) {
-      border: 10px solid black;
-      border-radius: 10% 10%;
-      box-shadow: 0px 0px 10px;
-    }
-
-    /* Offset the image to account for the border in the snapshot. */
-    ::shared-old(header-id) {
-      top: -10px;
-      left: -10px;
-    }
-  </style>
-</head>
-<body>
-  <div class="animated" sharedid="header-id" id="header">Shared Element</div>
-</body>
-</html>
 ```
+transition part container
+└─ transition part
+   ├─ texture (Page-A)
+   └─ texture (Page-B)
+```
+
+This allows for the container to be moved as one, while cross-fading the Page-A and Page-B content. The developer will also have access to both computed styles of the container for both Page-A and Page-B, so those can also be transitioned.
+
+Parts don't need to be associated with another part, which allows for transitions involving elements that are only in Page-A or only in Page-B.
+
+The root parts of each page are automatically associated.
 
-### New Document
+## Part 3: The transition
+
+Everything is now in place to perform the transition. The developer can move the parts around using the usual APIs, such as CSS and web animations.
+
+https://user-images.githubusercontent.com/93594/141100217-ba1fa157-cd79-4a9d-b3b4-67484d3c7dbf.mp4
+
+## Part 4: The end
+
+When the transition is complete, the top level is removed, revealing the real Page-B.
+
+# The API
+
+There are a lot of open questions around the API, but this section will give a flavour of the direction.
+
+Little thought has been giving to naming in these APIs, which will be improved when it's clearer what the right direction is.
+
+Additionally, this API focused on cross-document transitions.
```

Comment:
```suggestion
Additionally, this API focused on cross-document transitions, since it is not currently possible with existing web APIs.
```

Reply:
I meant that the SPA stuff is in a later section. I'll make this clearer.